src/com/palmergames/bukkit/towny/TownySettings.java

```diff
@@ -1947,20 +1962,27 @@ public static double getNationUpkeep() {
 	}
 
 	public static double getNationUpkeepCost(Nation nation) {
+		NationUpkeepCalculationEvent event = new NationUpkeepCalculationEvent(nation, getNationUpkeepCostRaw(nation));
+		Bukkit.getPluginManager().callEvent(event);
+		return event.getUpkeep();
+	}
+
+	private static double getNationUpkeepCostRaw(Nation nation) {
 
 		double multiplier;
 
 		if (nation != null) {
 			if (isNationUpkeepPerTown()) {
 				if (isNationLevelModifierAffectingNationUpkeepPerTown())
-					return (getNationUpkeep() * nation.getTowns().size()) * Double.valueOf(getNationLevel(nation).get(TownySettings.NationLevel.UPKEEP_MULTIPLIER).toString());
+					return (getNationUpkeep() * nation.getTowns().size()) * Double.parseDouble(getNationLevel(nation).get(TownySettings.NationLevel.UPKEEP_MULTIPLIER).toString());
 				else
 					return (getNationUpkeep() * nation.getTowns().size());
 			} else {
-				multiplier = Double.valueOf(getNationLevel(nation).get(TownySettings.NationLevel.UPKEEP_MULTIPLIER).toString());
+				multiplier = Double.parseDouble(getNationLevel(nation).get(TownySettings.NationLevel.UPKEEP_MULTIPLIER).toString());
 			}
-		} else
+		} else {
 			multiplier = 1.0;
```

Comment:
Can just initialize `multiplier` to `1.0` in the beginning to avoid this. Doesn't really make it less readable.